dfx/index.ts

```diff
@@ -60,8 +60,38 @@ export function getIdentityName(): string {
     return execSync(`dfx identity whoami`).toString().trim();
 }
 
-export function generateIdentity(name: string) {
-    execSync(`dfx identity new ${name} --storage-mode plaintext`);
+type StorageMode = 'keyring' | 'password-protected' | 'plaintext';
+
+export function generateIdentity(
+    name: string,
+    storageModeCool?: StorageMode
+): Buffer {
+    console.info();
+    console.info(`Generating identity ${name}`);
+    const storageMode = determineStorageMode(storageModeCool);
+    if (storageMode === undefined) {
+        console.info(`You may be prompted to create a password for ${name}`);
+        console.info();
+        return execSync(`dfx identity new ${name}`, {
+            stdio: ['inherit', 'pipe', 'inherit']
+        });
+    }
+    if (storageMode === 'password-protected') {
+        console.info(`You will be prompted to create a password for ${name}`);
+        console.info();
+    }
+    return execSync(`dfx identity new ${name} --storage-mode ${storageMode}`, {
+        stdio: ['inherit', 'pipe', 'inherit'] // TODO add todo here and aboe
+    });
+}
+
+function determineStorageMode(
+    storageMode?: StorageMode
+): StorageMode | undefined {
+    if (process.env.AZLE_PLAINTEXT_IDENTITY === 'true') {
+        return 'plaintext';
+    }
+    return storageMode;
 }
```

Comment:
When do we ever use the use or remove identity functions? This seems scary and hopefully never used, I wonder if we should remove these

Reply:
I had asked about this last PR, where this was introduced and we had decided to just leave them in. If we decide that we want to overturn that decisions I will remove it, but we should probably talk about why we are changing our mind about it

Reply:
get rid of use identity